android/libraries/rib-base/src/main/kotlin/com/uber/rib/core/Router.kt

```diff
@@ -157,9 +165,16 @@ protected constructor(
           .handleNonFatalWarning("A RIB tried to detach a child that was never attached", null)
       }
     }
-    childRouter.dispatchDetach()
+
+    triggerRibActionAndEmitEvents(
+      { childRouter.dispatchDetach() },
```

Comment:
This would probably be better as the leading parameter, so we can use trailing lambda syntax

Reply:
good idea, will update